src/swap/getExchangeRates.js

```diff
@@ -40,11 +43,17 @@ const getExchangeRates: GetExchangeRates = async (
   return res.data.map(
     ({ rate, rateId, provider, minAmountFrom, maxAmountFrom }) => {
       if (!rate || !rateId) {
-        throw new SwapExchangeRateOutOfBounds(null, {
-          unit: unitFrom.code,
-          minAmountFrom,
-          maxAmountFrom,
-        });
+        const isTooSmall = BigNumber(apiAmount).lt(minAmountFrom);
+
+        throw isTooSmall
+          ? new SwapExchangeRateAmountTooLow(null, {
+              unit: unitFrom.code,
+              minAmountFrom,
+            })
+          : new SwapExchangeRateAmountTooHigh(null, {
+              unit: unitFrom.code,
+              maxAmountFrom,
+            });
```

Comment:
how can we be sure the else case is a too high case? i'm a bit lost by this because it's the `!rate` case, there is no other possible issue?

Reply:
If you look at the api definition, it's either a rate or a bad amount response on this endpoint https://github.com/LedgerHQ/swap/blob/master/doc/openapi-swap.yaml#L49 so that's what I'm basing the code logic on.